src/Foundation/ErrorHandling/FrontendFormatter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation\ErrorHandling;
+
+use Flarum\Frontend\Controller;
+use Flarum\Http\Content\NotAuthenticated;
+use Flarum\Http\Content\NotFound;
+use Flarum\Http\Content\PermissionDenied;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Container\Container;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * A formatter for turning caught exceptions into "pretty" HTML error pages.
+ *
+ * For certain known error types, we display pages with dedicated information
+ * relevant to this class of error, e.g. a page with a search form for HTTP 404
+ * "Not Found" errors. We look for templates in the `views/error` directory.
+ *
+ * If no specific template exists, a generic "Something went wrong" page will be
+ * displayed, optionally enriched with a more specific error message if found in
+ * the translation files.
+ */
```

Comment:
Given that this behaves differently than the `ViewFormatter`, it should likely have a different docblock. :wink: